x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -396,8 +402,11 @@ export function deleteFeatureFromIndex(featureId: string) {
       return;
     }
     try {
+      dispatch(updateEditShape(DRAW_SHAPE.WAIT));
       await (layer as IVectorLayer).deleteFeature(featureId);
-      await dispatch(syncDataForLayerDueToDrawing(layer));
+      await dispatch(syncDataForLayerDueToDrawing(layer)).then(() => {
```

Comment:
Can you move the `.then` content to just be on the next line. Since the method is async and there is an await, `.then` is not needed and makes reading this more difficult and is not in the same style as everything else in this file